src/query/storages/system/src/query_cache_table.rs

```diff
@@ -0,0 +1,125 @@
+use common_meta_app::schema::TableMeta;
+use common_storages_result_cache::ResultCacheMetaManager;
+use common_storages_result_cache::RESULT_CACHE_PREFIX;
+use common_users::UserApiProvider;
+use itertools::Itertools;
+
+use crate::table::AsyncOneBlockSystemTable;
+use crate::table::AsyncSystemTable;
+
+pub struct QueryCacheTable {
+    table_info: TableInfo,
+}
+
+#[async_trait::async_trait]
+impl AsyncSystemTable for QueryCacheTable {
+    const NAME: &'static str = "system.query_cache";
+
+    fn get_table_info(&self) -> &TableInfo {
+        &self.table_info
+    }
+
+    async fn get_full_data(&self, ctx: Arc<dyn TableContext>) -> Result<DataBlock> {
+        let meta_client = UserApiProvider::instance().get_meta_store_client();
+        let result_cache_mgr = ResultCacheMetaManager::create(meta_client, 0);
+        let tenant = ctx.get_tenant();
+        let prefix = self.gen_result_cache_prefix(&tenant);
+
+        let cached_values = result_cache_mgr.list(prefix.as_str()).await?;
+
+        let mut sql_vec: Vec<&str> = Vec::with_capacity(cached_values.len());
+        let mut result_size_vec = Vec::with_capacity(cached_values.len());
+        let mut num_rows_vec = Vec::with_capacity(cached_values.len());
+        let mut partitions_sha_vec = Vec::with_capacity(cached_values.len());
+        let mut locations_vec = Vec::with_capacity(cached_values.len());
+
+        cached_values.iter().for_each(|x| {
+            sql_vec.push(x.sql.as_str());
+            result_size_vec.push(x.result_size as u64);
+            num_rows_vec.push(x.num_rows as u64);
+            partitions_sha_vec.push(x.partitions_shas.clone());
+            locations_vec.push(x.location.as_str());
+        });
+
+        let partitions_sha_vec: Vec<String> = partitions_sha_vec
+            .into_iter()
+            .map(|part| part.into_iter().join(", "))
+            .collect();
+
+        Ok(DataBlock::new_from_columns(vec![
+            StringType::from_data(sql_vec),
+            UInt64Type::from_data(result_size_vec),
+            UInt64Type::from_data(num_rows_vec),
+            StringType::from_data(
+                partitions_sha_vec
+                    .iter()
+                    .map(|part_sha| part_sha.as_str())
+                    .collect::<Vec<_>>(),
+            ),
+            StringType::from_data(locations_vec),
+        ]))
+    }
+}
+
+impl QueryCacheTable {
+    pub fn create(table_id: u64) -> Arc<dyn Table> {
+        let schema = TableSchemaRefExt::create(vec![
+            TableField::new("sql", TableDataType::String),
+            TableField::new("result_size", TableDataType::Number(NumberDataType::UInt64)),
+            TableField::new("num_rows", TableDataType::Number(NumberDataType::UInt64)),
+            TableField::new(
+                "partitions_sha",
+                TableDataType::Array(Box::new(TableDataType::String)),
+            ),
+            TableField::new("locations", TableDataType::String),

```

Comment:
Each cache only has one location now, so I think the table field name should be 'location'.